src/main/java/com/android/ex/chips/RecipientEditTextView.java

```diff
@@ -2492,51 +2470,59 @@ public void afterTextChanged(Editable s) {
         }
 
         @Override
-        public void onTextChanged(CharSequence s, int start, int before, int count) {
+        public void onTextChanged(CharSequence s, int start, int before, final int count) {
             // The user deleted some text OR some text was replaced; check to
             // see if the insertion point is on a space
             // following a chip.
             if (before - count == 1) {
-                // If the item deleted is a space, and the thing before the
-                // space is a chip, delete the entire span.
-                int selStart = getSelectionStart();
-                DrawableRecipientChip[] repl = getSpannable().getSpans(selStart, selStart,
-                        DrawableRecipientChip.class);
-                if (repl.length > 0) {
+                // If the selection was after a chip before the text changed, the user's intention is to delete the chip
+                if (mAfterChip != null && !mSkipRemove) {
                     // There is a chip there! Just remove it.
                     Editable editable = getText();
                     // Add the separator token.
-                    int tokenStart = mTokenizer.findTokenStart(editable, selStart);
+                    int tokenStart = mTokenizer.findTokenStart(editable, getSelectionStart());
                     int tokenEnd = mTokenizer.findTokenEnd(editable, tokenStart);
                     tokenEnd = tokenEnd + 1;
                     if (tokenEnd > editable.length()) {
                         tokenEnd = editable.length();
                     }
-                    if (isSuppliedContact(repl[0])) {
+                    if (isSuppliedContact(mAfterChip)) {
                         // In case of a generated contact remove the entire contact
-                        tokenStart = getSpannable().getSpanStart(repl[0]);
+                        tokenStart = getSpannable().getSpanStart(mAfterChip);
                     }
                     editable.delete(tokenStart, tokenEnd);
-                    getSpannable().removeSpan(repl[0]);
+                    getSpannable().removeSpan(mAfterChip);
                 }
+
+                mSkipRemove = false;
             } else if (count > before) {
-                if (mSelectedChip != null
-                    && isGeneratedContact(mSelectedChip)) {
+                int selStart = getSelectionStart();
+                if (selStart - count > 0 && selStart < getChipEnd(getLastChip())) {
+                    mSkipRemove = true;
+
+                    CharSequence typed = getText().subSequence(selStart - count, selStart);
+                    getText().delete(selStart - count, selStart);
+                    setSelection(getText().length());
+                    getText().append(typed);
```

Comment:
This is the only "hack" necessary when the selection is allowed to place itself naturally. When the user types text, we want to move it to the end. Also set mSkipRemove to skip the above code that deletes chips when you backspace over a single character following a chip.